libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -437,16 +437,14 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 		type = TypeProvider::withLocation(ref, typeLoc, isPointer);
 	}
 
-	if (
-		_variable.isConstant() &&
-		!dynamic_cast<UserDefinedValueType const*>(type) &&
-		type->containsNestedMapping()
-	)
-		m_errorReporter.fatalDeclarationError(
-			3530_error,
-			_variable.location(),
-			"The type contains a (nested) mapping and therefore cannot be a constant."
-		);
+	if (_variable.isConstant() && !type->isValueType())
+	{
+		bool allowed = false;
+		if (auto arrayType = dynamic_cast<ArrayType const*>(type))
+			allowed = arrayType->isByteArray();
+		if (!allowed)
+			m_errorReporter.fatalTypeError(9259_error, _variable.location(), "Only constants of value type and byte arrays are implemented.");
```

Comment:
If we want to be pedantic this should probably either be
``Only constants of value and byte array type...``, resp. ``Only constants of value type and byte array type...``
or
``Only constants of value type and byte array constants...``